### PR TITLE
Add more generic names for classes and variables

### PR DESCRIPTION
**Pull request by viktoria-mahmud**

### CODE REVIEW

**Review by Cosmin-Daniel — APPROVED** [review]

File: source/_patterns/01-molecules/navigation/view-selector.mustache [hunk 1]
```diff
@@ -1,13 +1,13 @@
-<div class="view-selector{{#figureUrl}} view-selector--has-figures{{/figureUrl}}" data-behaviour="ViewSelector"{{#sideBySideUrl}} data-side-by-side-link="{{sideBySideUrl}}"{{/sideBySideUrl}}>
+<div class="view-selector{{#secondaryUrl}} view-selector--has-secondary{{/secondaryUrl}}" data-behaviour="ViewSelector"{{#sideBySideUrl}} data-side-by-side-link="{{sideBySideUrl}}"{{/sideBySideUrl}}>
   <ul class="view-selector__list">
-    <li class="view-selector__list-item view-selector__list-item--article{{^figureIsActive}} view-selector__list-item--active{{/figureIsActive}}">
-      <a href="{{articleUrl}}" class="view-selector__link view-selector__link--article"><span>Article</span></a>
+    <li class="view-selector__list-item view-selector__list-item--main{{^secondaryIsActive}} view-selector__list-item--active{{/secondaryIsActive}}">
+      <a href="{{mainUrl}}" class="view-selector__link view-selector__link--main"><span>Article</span></a>
```
Comment (nlisgo) [inline source/_patterns/01-molecules/navigation/view-selector.mustache:4]: If we intend to use the view-selector template do we need to also add a parameter for `mainUrlLabel` and `secondaryUrlLabel`?

File: source/_patterns/01-molecules/navigation/view-selector.yaml [hunk 2]
```diff
@@ -4,13 +4,13 @@ properties:
   sideBySideUrl:
     type: string
     minLength: 1
-  articleUrl:
+  mainUrl:
```
Comment (nlisgo) [inline source/_patterns/01-molecules/navigation/view-selector.yaml:7]: I think perhaps this should be `primaryUrl`.

File: source/_patterns/01-molecules/navigation/view-selector.mustache (hunk 1 above)
Comment (viktoria-mahmud) [reply source/_patterns/01-molecules/navigation/view-selector.mustache:4]: Thanks, it is part of the [PR](https://github.com/elifesciences/pattern-library/pull/1178) for Tabs in Home page but it is better to be part of this one

Comment (viktoria-mahmud) [reply source/_patterns/01-molecules/navigation/view-selector.mustache]: Done, hardcoded text is removed

File: source/_patterns/01-molecules/navigation/view-selector.yaml (hunk 2 above)
Comment (viktoria-mahmud) [reply source/_patterns/01-molecules/navigation/view-selector.yaml]: All variables that contain "main" are changed to "primary"

File: source/_patterns/01-molecules/navigation/view-selector.yaml [hunk 3]
```diff
@@ -4,15 +4,21 @@ properties:
...
     minLength: 1
-  figureUrl:
+  secondaryUrl:
     type: string
     minLength: 1
-  figureIsActive:
+  secondaryIsActive:
     type: boolean
     default: false
+  primaryViewSelector:
```
Comment (nlisgo) [inline source/_patterns/01-molecules/navigation/view-selector.yaml:16]: This would be better as `PrimaryLabel` and will also need to be required.

Comment (viktoria-mahmud) [reply source/_patterns/01-molecules/navigation/view-selector.yaml]: Done, primaryViewSelector is renamed to primaryLabel and it is required, secondaryViewSelector is renamed to secondaryLabel

File: source/_patterns/01-molecules/navigation/view-selector.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
-  "articleUrl": "#article",
-  "figureUrl": "#figures",
+  "primaryUrl": "#article",
+  "secondaryUrl": "#figures",
```
Comment (nlisgo) [inline source/_patterns/01-molecules/navigation/view-selector.json:3]: Need to add `primaryLabel` and `secondaryLabel` to this default json example.

**Review by nlisgo — APPROVED** [review]

**Comment by nlisgo** [comment]
@viktoria-mahmud if you can just verify that you are ok with the few recent changes I've made. Happy to discuss any of them. If you are ok we can merge in the morning.

@Cosmin-Daniel I can offer some support on the patterns-php integration.

**Comment by viktoria-mahmud** [comment]
> @viktoria-mahmud if you can just verify that you are ok with the few recent changes I've made. Happy to discuss any of them. If you are ok we can merge in the morning.

I am happy with these changes, thank you!

**Comment by nlisgo** [comment]
Because we are changing an existing template we should probably get a head start on the patterns-php and journal PR's before we merge this in. If we merge this in but there is a delay on the patterns-php and journal then we may block other work getting into the journal.

**Pull request merged**